internal/pkg/agent/storage/store/internal/migrations/migrations.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package migrations
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"gopkg.in/yaml.v2"
+)
+
+type StateStore struct {
+	Action      Action        `yaml:"action"`
+	AckToken    string        `yaml:"ack_token"`
+	ActionQueue []ActionQueue `yaml:"action_queue"`
+}
+
+// Action is a struct to read an Action Policy Change from its old
+// YAML format.
+type Action struct {
```

Comment:
There is also a private `type actionQueue []fleetapi.ScheduledAction` in state_store.go that makes following this even more confusion.

Why isn't this the general `fleetapi.Action`? Or one of the concrete Fleet action API implementations?

Reply:
This package is only for action and state store migrations. It cannot be imported outside the `strore` package. That's why I thought it would not be confusing, at this level there is only one meaning for `Action`.

>Why isn't this the general `fleetapi.Action`? Or one of the concrete Fleet action API implementations?

because it's a code for migration, the `fleetapi` models do not work, they changed. There was a whole PR just to fix the schema: https://github.com/elastic/elastic-agent/pull/4240

Reply:
Also it was copied from the old store code. Even thought it has properties to read different actions, as far as I can tell, there should only be 'Policy Change' actions saved